docs to s3 bucket

mikestonecodes opened the pull request:
<!--
  Thank you for your pull request. Please provide a description above and review
  the requirements below.

  Contributors guide: https://github.com/MARKETProtocol/meta/blob/master/CONTRIBUTING.md
-->

##### Description
<!-- A description on what this PR aims to solve -->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] Linter status: 100% pass
- [x] Changes don't break existing behavior
- [x] Tests coverage hasn't decreased

##### Affected core subsystem(s)
<!-- Provide affected core subsystem(s) (like Docs, UI, UX, Tests etc). -->

##### Testing
<!-- Why should the PR reviewer trust that this change doesn't break anything? How have you tested this change? -->

##### Refers/Fixes
<!--
  Link to an issue if applicable. For example:
  If your PR fixes an issue -- Fixes: #102
  If your PR refers an issue -- Refs: #101
-->
#105 

Automatic doc deployment to S3 Bucket. Uses environment variables  ( can be set with .env file )

AWS_ACCESS_KEY
AWS_SECRET
AWS_REGION
AWS_BUCKET
DOC_TMP_DIR ( if null /tmp/docs-aws )

1) checks out MARKET.js.git 
2) goes through each tag / version and copies 'doc folder' to tmp folder
3) checks out  docs.git  , copies gh-pages branch into same tmp folder
4) uploads to S3 Bucket

triggered in travis 

example:
https://s3.eu-central-1.amazonaws.com/marketjstest/index.html
https://s3.eu-central-1.amazonaws.com/marketjstest/0.2.2/index.html

THIS WILL FAIL RIGHT NOW

needed on deployment server  : environment variables set for correct s3 bucket key etc  ( see above)

tsc installed globally
node installed globally

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/MARKETProtocol/MARKET.js?pullRequest=142) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/MARKETProtocol/MARKET.js?pullRequest=142) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/MARKETProtocol/MARKET.js?pullRequest=142) it.</sub>

pelsasser commented:
@mikestonecodes - before we merge this, can we remove the logging of the credentials?  Once that is done, I will go ahead and merge and attempt to get travis all squared with the needed env variables to work.

pelsasser commented on tools/update-docs.js:
```diff
@@ -0,0 +1,61 @@
+exec("git init");
+exec("git clone  https://github.com/MARKETProtocol/MARKET.js.git ");
+cd("MARKET.js");
+console.log(process.env.AWS_ACCESS_KEY, process.env.AWS_SECRET, process.env.AWS_REGION);
```
Don't think this is a good idea ;) can we remove!

mikestonecodes replied:
LOL whoops.

pelsasser commented on tools/update-docs.ts:
```diff
@@ -0,0 +1,81 @@
+exec(`git clone  https://github.com/MARKETProtocol/MARKET.js.git `);
+cd("MARKET.js")
+
+console.log( process.env.AWS_ACCESS_KEY, process.env.AWS_SECRET, process.env.AWS_REGION)
```
here too.

pelsasser commented:
@mikestonecodes - I have env variables set, bucket created, etc.

Node should be good to go, we may need to modify travis for tsc install.

mikestonecodes commented:
hey here are the changes I made
1) removed dependency on tsc 
2) removed console logs ( lol sorry )
3) removed retrieving docs repo in favor of this https://github.com/MARKETProtocol/docs/pull/29

pelsasser commented:
@mikestonecodes looks like this is failing -  https://travis-ci.org/MARKETProtocol/MARKET.js/builds/418842684#L1814
